### PR TITLE
device tracker battery level badge

### PR DESCRIPTION
**Pull request by iantrich**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

image

### CODE REVIEW

File: src/components/entity/ha-state-label-badge.ts [hunk 1]
```diff
@@ -166,6 +168,31 @@ export class HaStateLabelBadge extends LitElement {
...
+    state: HassEntity
+  ): string | undefined {
+    if (["device_tracker"].includes(domain) && state.attributes.battery) {
+      return state.attributes.battery + "%";
+    }
+
+    return undefined;
+  }
+
+  private _computeUpperLabelIcon(
```
Comment (iantrich) [inline src/components/entity/ha-state-label-badge.ts:182]: I played around a bit with an icon in the label and haven't decided if it looks good or not:
image
I think it takes away more than it adds though

File: src/components/entity/ha-state-label-badge.ts [hunk 2]
```diff
@@ -166,6 +168,31 @@ export class HaStateLabelBadge extends LitElement {
...
+    domain: string,
+    state: HassEntity
+  ): string | undefined {
+    if (["device_tracker"].includes(domain) && state.attributes.battery) {
+      const battery = Math.round(state.attributes.battery / 10) * 10;
```
Comment (zsarnett) [inline src/components/entity/ha-state-label-badge.ts:187]: Does rounding the attr not work here? why divide and then multiply

File: src/components/ha-label-badge.ts [hunk 3]
```diff
@@ -25,11 +27,23 @@ class HaLabelBadge extends LitElement {
     return html`
       <div class="badge-container">
         <div class="label-badge" id="badge">
+          ${this.upperLabel
+            ? html`
+                <div
+                  class=${classMap({
+                    upperLabel: true,
+                    big: this.upperLabel.length > 5,
+                  })}
```
Comment (zsarnett) [inline src/components/ha-label-badge.ts:36]:
```suggestion
                  class="upperLabel ${classMap({
                    big: this.upperLabel.length > 5,
                  })}"
```

File: src/components/ha-label-badge.ts [hunk 4]
```diff
@@ -25,11 +27,23 @@ class HaLabelBadge extends LitElement {
...
+                >
+                  <span>${this.upperLabel}</span>
+                </div>
+              `
+            : ""}
           <div
-            class="${classMap({
+            class=${classMap({
               value: true,
```
Comment (zsarnett) [inline src/components/ha-label-badge.ts:44]: you didnt write this but value: true can just be the string in the class like above

File: src/components/ha-label-badge.ts [hunk 5]
```diff
@@ -43,10 +57,10 @@ class HaLabelBadge extends LitElement {
           ${this.label
             ? html`
                 <div
-                  class="${classMap({
+                  class=${classMap({
                     label: true,
```
Comment (zsarnett) [inline src/components/ha-label-badge.ts:61]: same

File: src/components/ha-label-badge.ts [hunk 6]
```diff
@@ -25,11 +27,23 @@ class HaLabelBadge extends LitElement {
...
         <div class="label-badge" id="badge">
+          ${this.upperLabel
+            ? html`
+                <div
+                  class=${classMap({
+                    upperLabel: true,
+                    big: this.upperLabel.length > 5,
+                  })}
+                >
+                  <span>${this.upperLabel}</span>
```
Comment (zsarnett) [inline src/components/ha-label-badge.ts:38]: I think is just similar to the rest but is the span needed?

File: src/components/entity/ha-state-label-badge.ts (hunk 2 above)
Comment (iantrich) [reply src/components/entity/ha-state-label-badge.ts:187]: I want to round to the nearest ten for the icon.  Probably won't keep this around, I don't think the icon adds much

File: src/components/ha-label-badge.ts (hunk 6 above)
Comment (iantrich) [reply src/components/ha-label-badge.ts:37]: Removing the span and moving the css to the div actually does make it look a bit different. I didn't want to bother with it for this and just use what is there.

**Comment by bramkragten** [comment]
I don't think the upper label is a good idea, we just don't have the space in a badge. I think this should be a card, something like the demo from Kernehed:

image

**Comment by zsarnett** [comment]
something like this maybe: https://github.com/home-assistant/frontend/issues/6578

**Comment by iantrich** [comment]
I'm not against a card, but wanted to focus on a badge design.
Maybe instead of displaying home/away/zone, we could instead display the icon on the bottom left and the battery percentage on the bottom right?

**Comment by bramkragten** [comment]
Can you show a mockup of what you mean? I think it is just too much information for a badge, or you would have to remove the image, but...

**Comment by iantrich** [comment]
I played around with it a bit tonight and couldn't get anything that worked any better than the first proposal. Closing.

**Pull request closed without merging**